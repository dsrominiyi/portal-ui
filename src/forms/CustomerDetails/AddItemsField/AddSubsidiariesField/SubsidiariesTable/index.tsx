import React, { FunctionComponent } from 'react';
import { utc as moment } from 'moment';

import TableDisplay from '../../../../../components/TableDisplay';

import styles from './styles.scss';

export const SubsidiariesTable: FunctionComponent<ItemTableProps> = ({
  items: subsidiaries,
  onEdit,
  onDelete
}: ItemTableProps) => {
  const rows = (subsidiaries as Company[]).map(
    ({ companyName, tradingName, typeStatus, tradingSinceDate, registeredCoNumber }) => [
      companyName,
      tradingName,
      typeStatus,
      tradingSinceDate ? moment(tradingSinceDate, 'YYYY-MM-DD').format('DD/MM/YYYY') : '',
      registeredCoNumber
    ]
  );

  return (
    <TableDisplay
      additionalClasses={[styles.subsidiariesTable]}
      headers={[
        'Company Name',
        'Trading Name',
        'Legal Status',
        'Trading Since',
        'Registration Number'
      ]}
      rows={rows}
      onEdit={onEdit}
      onDelete={onDelete}
    />
  );
};

SubsidiariesTable.displayName = 'SubsidiariesTable';
export default SubsidiariesTable;
