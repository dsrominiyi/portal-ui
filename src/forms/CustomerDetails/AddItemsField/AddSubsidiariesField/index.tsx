import React, { FunctionComponent } from 'react';

import AddItemsField from '..';
import SubsidiaryForm from '../../SubsidiaryForm';
import SubsidiariesTable from './SubsidiariesTable';

interface Props {
  subsidiaries?: Company[];
  onChange(values: Company[]): void;
}

export const AddSubsidiariesField: FunctionComponent<Props> = ({
  subsidiaries,
  onChange
}: Props) => (
  <AddItemsField
    fieldText="Subsidiaries (optional)"
    itemName="subsidiary"
    form={SubsidiaryForm}
    table={SubsidiariesTable}
    items={subsidiaries}
    onChange={onChange}
  />
);

AddSubsidiariesField.displayName = 'AddSubsidiariesField';
export default AddSubsidiariesField;
