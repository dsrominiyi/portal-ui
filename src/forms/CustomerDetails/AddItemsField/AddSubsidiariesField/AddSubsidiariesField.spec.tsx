import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import AddSubsidiariesField from '.';

let component: ShallowWrapper;
const onChange = jest.fn();

describe('AddSubsidiariesField', () => {
  test('AddSubsidiariesField matches snapshot', () => {
    component = shallow(<AddSubsidiariesField onChange={onChange} />);
    expect(component).toMatchSnapshot();
  });
});
