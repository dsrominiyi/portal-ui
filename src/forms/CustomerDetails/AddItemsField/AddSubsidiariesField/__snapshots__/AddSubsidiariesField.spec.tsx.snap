// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSubsidiariesField AddSubsidiariesField matches snapshot 1`] = `
<AddItemsField
  fieldText="Subsidiaries (optional)"
  form={[Function]}
  itemName="subsidiary"
  onChange={[MockFunction]}
  table={[Function]}
/>
`;
